pymc3/plots/densityplot.py

```diff
@@ -145,34 +148,43 @@ def _kde_helper(vec, vname, c, bw, alpha, point_estimate, hpd_markers, outline,
         (opaque). Defaults to 0.
     ax : matplotlib axes
     """
-    density, l, u = fast_kde(vec, bw)
-    x = np.linspace(l, u, len(density))
-    hpd_ = hpd(vec, alpha)
-    cut = (x >= hpd_[0]) & (x <= hpd_[1])
-
-    xmin = x[cut][0]
-    xmax = x[cut][-1]
-    ymin = density[cut][0]
-    ymax = density[cut][-1]
-
-    if outline:
-        ax.plot(x[cut], density[cut], color=c)
-        ax.plot([xmin, xmin], [-0.5, ymin], color=c, ls='-')
-        ax.plot([xmax, xmax], [-0.5, ymax], color=c, ls='-')
+    if vec.dtype.kind == 'f':
+        density, l, u = fast_kde(vec)
+        x = np.linspace(l, u, len(density))
+        hpd_ = hpd(vec, alpha)
+        cut = (x >= hpd_[0]) & (x <= hpd_[1])
+
+        xmin = x[cut][0]
+        xmax = x[cut][-1]
+        ymin = density[cut][0]
+        ymax = density[cut][-1]
+
+        if outline:
+            ax.plot(x[cut], density[cut], color=c)
+            ax.plot([xmin, xmin], [-ymin/100, ymin], color=c, ls='-')
+            ax.plot([xmax, xmax], [-ymax/100, ymax], color=c, ls='-')
+
+        if shade:
+            ax.fill_between(x, density, where=cut, color=c, alpha=shade)
+
+    else:
+        xmin, xmax = hpd(vec, alpha)
+        bins = range(xmin, xmax+1)
+        if outline:
+            ax.hist(vec, bins=bins, color=c, histtype='step')
+        ax.hist(vec, bins=bins, color=c, alpha=shade)
```

Comment:
Better to set `density=None` to True.

Reply:
That will only change the scale of y. But that scale is not plotted, the yticks are set to an empty list.

Reply:
I see. Fine with me.